vscode/src/definition.ts

```diff
@@ -23,10 +24,28 @@ class QSharpDefinitionProvider implements vscode.DefinitionProvider {
     );
     if (!definition) return null;
     const uri = vscode.Uri.parse(definition.source);
-    // We have to do this to map the position :(
-    const definitionPosition = (
-      await vscode.workspace.openTextDocument(uri)
-    ).positionAt(definition.offset);
+
+    const definitionPosition = (await openDocument(uri)).positionAt(
+      definition.offset,
+    );
     return new vscode.Location(uri, definitionPosition);
   }
 }
+
+async function openDocument(uri: vscode.Uri) {
+  // Search both text documents and notebook cell documents
+  const doc = vscode.workspace.textDocuments
+    .concat(
+      vscode.workspace.notebookDocuments.flatMap((doc) =>
+        doc
+          .getCells()
+          .filter((cell) =>
+            vscode.languages.match(qsharpDocumentFilter, cell.document),
+          )
+          .map((cell) => cell.document),
+      ),
+    )
+    .find((doc) => doc.uri.toString() === uri.toString());
+
+  return doc || (await vscode.workspace.openTextDocument(uri));
+}
```

Comment:
Does it make sense to cache the uri.toString() so that we aren't calling it for every comparison?

Reply:
It does. Thanks!